cmd/syft/cli/attest/attest.go

```diff
@@ -183,23 +209,72 @@ func generateAttestation(predicate []byte, src *source.Source, sv *sign.SignerVe
 		return errors.Wrap(err, "unable to sign SBOM")
 	}
 
+	if app.Attest.NoUpload {
+		bus.Publish(partybus.Event{
+			Type: event.Exit,
+			Value: func() error {
+				_, err := os.Stdout.Write(signedPayload)
+				return err
+			},
+		})
+	}
+
+	return uploadAttestation(app, signedPayload, digest, sv)
+}
+
+func uploadAttestation(app *config.Application, signedPayload []byte, digest name.Digest, sv *sign.SignerVerifier) error {
+	opts := []static.Option{static.WithLayerMediaType(types.DssePayloadType)}
+	if sv.Cert != nil {
+		opts = append(opts, static.WithCertChain(sv.Cert, sv.Chain))
+	}
+
+	bundle, err := uploadToTlog(context.TODO(), sv, app.Attest.RekorURL, func(r *client.Rekor, b []byte) (*models.LogEntryAnon, error) {
+		return cosign.TLogUploadInTotoAttestation(context.TODO(), r, signedPayload, b)
+	})
+	if err != nil {
+		return err
+	}
+	opts = append(opts, static.WithBundle(bundle))
+	sig, err := static.NewAttestation(signedPayload, opts...)
+	if err != nil {
+		return err
+	}
+
+	se, err := ociremote.SignedEntity(digest)
+	if err != nil {
+		return err
+	}
+
+	// Attach the attestation to the entity.
+	newSE, err := mutate.AttachAttestationToEntity(se, sig)
+	if err != nil {
+		return err
+	}
+
+	// Publish the attestations associated with this entity
+	err = ociremote.WriteAttestations(digest.Repository, newSE)
```

Comment:
is there a way to get observability into the upload and publish that to the bus for the ETUI to pick up on?

Reply:
That's a good question. I don't think we have access to things within `WriteAttestation`, but let me see what we can pull off of newSE to publish. Right now we only publish the payload. This is the field that gets revealed when calling `COSIGN_EXPERIMENTAL=1 cosign verify-attestation` so I think it's the most important bit to report on during publish to the ETUI.

Reply:
Talked offline about these comments - this is less about showing more information and instead adding interaction to the ETUI that displays progress and steps surrounding the attestation uploads